operator-common/src/main/java/io/strimzi/operator/common/operator/resource/StatusUtils.java

```diff
@@ -100,19 +100,30 @@ public static Condition buildRebalanceCondition(String type) {
         if (resource.getMetadata().getGeneration() != null)    {
             status.setObservedGeneration(resource.getMetadata().getGeneration());
         }
+        // condition contains a timestamp so using a set does not help
         List<Condition> conditionList = new ArrayList<>();
         Condition condition = StatusUtils.buildCondition(type, conditionStatus, null);
         conditionList.add(condition);
 
         if (status.getConditions() != null) {
             status.getConditions().forEach(cond -> {
                 if ("UnknownFields".equals(cond.getReason()) || "DeprecatedFields".equals(cond.getReason())) {
-                    conditionList.add(cond);
+                    if (!conditionAlreadyPresent(conditionList, cond)) {
+                        conditionList.add(cond);
+                    }
                 }
             });
         }
+        status.setConditions(new ArrayList<>(conditionList));
+    }
 
-        status.setConditions(conditionList);
+    private static boolean conditionAlreadyPresent(List<Condition> conditions, Condition condition) {
+        return conditions.stream().filter(cond -> {
+            return condition.getReason().equals(cond.getReason()) &&
+                    condition.getType().equals(cond.getType()) &&
+                    condition.getStatus().equals(cond.getStatus()) &&
+                    condition.getMessage().equals(cond.getMessage());
```

Comment:
isn't it enough to filter on type? That should be the "key", or?

Reply:
Type is for example a `warning`. So using it as a key would result to having just one warning even in the case when there are more ones. I think more suitable one is `reason`.

Reply:
Yes maybe